### PR TITLE
<amp-lightbox-gallery> validator rules

### PR DESCRIPTION
**Pull request by cathyxz**
Adds validator rules for `<amp-lightbox-gallery>`. I still need to add one more validator rule that specifies that only children of `<amp-carousel>` with the `lightbox` attribute can include the attribute `lightbox-exclude`. Because validator is so fickle with regards to whitespace, I'm opening a second PR for it. 

Partially addresses https://github.com/ampproject/amphtml/issues/13789

### CODE REVIEW

File: extensions/amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.out [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+|  
+|    <!-- Invalid: missing value for lightbox-thumbnail-id -->
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox lightbox-thumbnail-id></amp-img>
+>>   ^~~~~~~~~
+amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.html:41:2 The attribute 'lightbox-thumbnail-id' in tag 'amp-img' is set to the invalid value ''. (see https://www.ampproject.org/docs/reference/components/amp-img) [AMP_TAG_PROBLEM]
+|    <!-- Invalid: missing lightbox attribute for element with lightbox-thumbnail-id -->
+|    <amp-img src="/awesome.png" width="300" height="300" lightbox-thumbnail-id="a"></amp-img>
+>>   ^~~~~~~~~
+amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.html:43:2 The attribute 'lightbox' in tag 'amp-img' is missing or incorrect, but required by attribute 'lightbox-thumbnail-id'. (see https://www.ampproject.org/docs/reference/components/amp-img) [DISALLOWED_HTML]
+|    <!-- Invalid: cannot have lightbox-exclude on an item that isn't in a carousel -->
```
Comment (honeybadgerdontcare) [inline extensions/amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.out]: Note that this isn't being marked as invalid. It seems like you want `lightbox-exclude` to only be an attribute allowed on `amp-carousel` children. The only way to enforce that currently would be to add it as an allowed attribute for all possible children tags of `amp-carousel`.

Comment (cathyxz) [reply extensions/amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.out]: Yup. Let me remove this test case for now. What I intend to do is you can only have `lightbox-exclude` on children of `<amp-carousel>` that have the `lightbox` attribute. I think the proper way to do this is to add a reference point in `<amp-carousel>` refering to `<amp-carousel>`s with the lightbox attribute, and then to add said reference point as a mandatory parent in `lightbox-exclude` under the main validator `.protoascii` file.

Comment (cathyxz) [reply extensions/amp-lightbox-gallery/0.1/test/validator-amp-lightbox-gallery.out]: Okay, it seems like we can't add `mandatory_parent` or `mandatory_ancestor` attributes to to `attr`. 

I tried adding a reference point to `<amp-carousel>` that applies to its children. However, this conflicts with the `<amp-selector>` reference points, which cover all children of `<amp-selectors>`.

File: validator/validator-main.protoascii [hunk 2]
```diff
@@ -4486,6 +4486,20 @@ attr_lists: {
     value_casei: "parallax"
     requires_extension: "amp-fx-collection"
   }
+  # amp-lightbox-gallery specific attributes, see
+  # https://www.ampproject.org/docs/reference/components/amp-lightbox-gallery
+  attrs: {
```
Comment (aghassemi) [inline validator/validator-main.protoascii]: to @honeybadgerdontcare points, yeah we don't want these on everything. Maybe makes sense to create an `attr_lists` for `lightboxable elements` (similar to how `extended-amp-global` works for common attributes like `layout`)  and add that to only components we currently support.

Comment (cathyxz) [reply validator/validator-main.protoascii]: +1. 
Refactored this to produce a `lightboxable-elements` `attr_list`, stuck on some whitespace validation issue again.

File: validator/validator-main.protoascii [hunk 3]
```diff
@@ -4084,6 +4084,24 @@ attr_lists: {
   attrs: { name: "media" }
   attrs: { name: "noloading" value: "" }
 }
+
+# Lightboxable Elements
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: Perhaps move into `validator-amp-lightbox-gallery.protoascii`?

Comment (cathyxz) [reply validator/validator-main.protoascii]: Good point!

**Pull request merged**